---
title: "PCA for Adaptation is RAD"
author: Angela Bunning
output: html_notebook
---
# README for generating first set of PCA plots in Funk *et al*, 2016.


### The files that were on the dryad repositiory for the paper are Genepop input files which contain each SNP by name and each indvidual from each island population used post filtering process. (See Funk *et al*, 2016 for more informating when it comes to filtering the SNPs and indviduals). These files were not read to put directly into `prcomp` so we used UNIX to format these files. 

### Description of files from Dryad repository:
SNP genotypes (GP_with_grays.txt and GP_NO_grays.txt)
-- These 2 Genepop input files give SNP genotypes with vs. without gray foxes included (see Funk et al. 2016 for details).  The first row is a file description, followed by locus (contig names) for variable loci remaining after filters, followed by genotypes for each individual. Allelic states are coded as: A = 1, C = 2, G = 3, T = 4. Some individuals were removed due to low coverage.

Individuals included in "GP_with_grays.txt":
- grays (individuals gray_1, gray_3-13, gray_15-18)
- smi (individuals smi_1-15, smi_17-19, smi_21, smi_23-24)
- sri (individuals sri_1-23)
- sci (individuals sci_1-24)
- sca (individuals sca-w_1-9, sca-w_11-18, sca-w_20-24, sca-e_1-10, sca-e_12-22)
- scl (individuals scl_2-4, scl_6-19)
- sni (individuals sni-redeye_1-20, sni-redeye_22-24, sni-skyline_1-10, sni-skyline_12-22)

Individuals included in "GP_NO_grays.txt":
- smi (individuals smi_1-19, smi_21, smi_23-24)
- sri (individuals sri_1-23)
- sci (individuals sci_1-24)
- sca (individuals sca-w_1-9, sca-w_11-18, sca-w_20-24, sca-e_1-22)
- scl (individuals scl_2-19)
- sni (individuals sni-redeye_1-20, sni-redeye_22-24, sni-skyline_1-10, sni-skyline_12-22)


## Genepop input file manipulation in UNIX  

Remove all SNP names becuase they are not needed for the PCA plot:

```{bash}
sed '1,4859d' allfoxes.txt # this removes the range of lines from 1-4859 which accounts for all the SNPs in the files 

sed '1,5294d' nogreyfoxes.txt # this removes the range of lines from 1-4859 which accounts for all the SNPs in the files 

# this function was taken from http://www.folkstalk.com/2013/03/sed-remove-lines-file-unix-examples.html 
```

The input files come with multiple delimiters so conver the files to a consitent one:
```{bash}
sed 's/,/\t/g' allfoxesnoSNPs.txt > allfoxesnoSNPstab.txt #for the file containing island foxes and gray fox populations

sed 's/,/\t/g' nogreyfoxes.txt #for the file containg just containing island fox populations

# this function was taken from http://www.theunixschool.com/2012/09/examples-how-to-change-delimiter-of-file-Linux.html
```

Remove the lines in the file that have the word 'pop' in them and put the lines of populations into another file:
```{bash}
grep -E 'grey|smi|sri|sci|sca|scl|sni' allfoxesnoSNPstab.txt > allfoxesnoSNPstabpop.txt 

grep -E 'grey|smi|sri|sci|sca|scl|sni' nogreyfoxesnoSNPstab.txt > nogreyfoxesnoSNPstabpop.txt

# This function was taken from http://www.thegeekstuff.com/2011/10/grep-or-and-not-operators
```


## Generating PCA plots in R
```{r}

allfoxes <- read.table(file = "foxes_goodfile.txt", header = FALSE) #load in file as a table
pops_grey <- allfoxes[,1] # subsets population data 
grey_foxes_pca <- prcomp(allfoxes[,2:4859], scale = TRUE) # runs PCA function on the subsetted data 
summary(grey_foxes_pca) # gives you stats for all of the componets and the percentage that each componet contributes to variation 
gf_pc_scores <- data.frame(grey_foxes_pca$x) # puts all the principle componets into a dataframe 
allfoxplot <- ggplot(gf_pc_scores) + geom_point(aes(x = PC1, y = PC3, colour = pops_grey)) + coord_equal() + ggtitle("PCA with Island Populations & Grey Fox") + labs(x = "PC1(18.3%)", y = "PC3(35.3%)") + theme(text = element_text(size = 20), panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), panel.background = element_blank(), legend.key = element_blank()) #plot function with ggplot 2 
allfoxplot + scale_color_brewer(palette = "Dark2")
```

```{r}
nogrey_foxes <- read.table(file = "no_gray_foxes_goodfile.txt", header = FALSE) #load in file as a table
pops_ng <- nogrey_foxes[,1] # subsets population data 
ng_foxes_pca <- prcomp(nogrey_foxes[,2:5294], scale = TRUE) # runs PCA function on the subsetted data 
summary(ng_foxes_pca) # gives you stats for all of the componets and the percentage that each componet contributes to variation 
ngf_pc_scores <- data.frame(ng_foxes_pca$x) # puts all the principle componets into a dataframe 
nogreyplot <- ggplot(ngf_pc_scores) + geom_point(aes(x = PC1, y = PC3, colour = pops_ng)) + coord_equal() + ggtitle("PCA with Island Populations") + labs(x = "PC1(21.1%)", y = "PC3(39.9%)") + theme(text = element_text(size = 20), panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"), panel.background = element_blank(), legend.key = element_blank()) #plotting function with ggplot2
nogreyplot + scale_color_brewer(palette = "Dark2")
```

